src/coreclr/minipal/Windows/dn-u16.cpp

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <Windows.h>
+#include <wchar.h>
+
+#include <dn-u16.h>
+
+size_t strlen_u16(const WCHAR* str)
```

Comment:
Can we use `char16_t` type instead of `WCHAR` in all this?

Reply:
Not yet. That is the longer term plan. I want to keep with `WCHAR` until we remove the PAL. Once we start creating a proper abstraction layer (i.e., `minipal`) that change will occur.